services/processor/native/call.go

```diff
@@ -52,23 +52,30 @@ func (s *service) verifyInternalMethodCall(contractInfo *types.ContractInfo, met
 	return errors.Errorf("internal method '%s' called from different service '%s' without system permissions", methodInfo.Name, callingService)
 }
 
-func (s *service) processMethodCall(contractInfo *types.ContractInfo, methodInfo *types.MethodInfo, args []*protocol.MethodArgument) ([]*protocol.MethodArgument, error, error) {
+func (s *service) processMethodCall(ctx types.Context, contractInfo *types.ContractInfo, methodInfo *types.MethodInfo, args []*protocol.MethodArgument) (outArgs []*protocol.MethodArgument, outErr error, err error) {
```

Comment:
why do we need two error return values?

Reply:
We differentiate between the contract inside returned an error to that there was an error running the contract (whether the error was in Orbs core code or the contract developer's code)

Two examples:

1. You try to read a value from StateStorage but the state DB is down so the call fails - this is an error - this is an Orbs core error

2. The contract transfer attempts to draw money from an account and it will bring the account to a negative balance, which throws an error in the contract - this is a contract developer error